arch/arm/src/samv7/sam_progmem.c

```diff
@@ -202,9 +202,9 @@ void sam_progmem_initialize(void)
 {
   uint32_t regval;
 
-  /* Set flash access mode to 128bit and wait status to 4 */
+  /* Flash access mode is 128bit always so configure wait states to 6 */
 
-  sam_eefc_initaccess(SAM_EFC_ACCESS_MODE_128, 4);
+  sam_eefc_initaccess(6);
```

Comment:
Suggestion: define a symbol with flash wait states equal 6 instead of hardcoding it in two places

Reply:
Ok. I will add this to Kconfig

Reply:
Done. Thank you @acassis for your comment. It lead me to re-evaluate solution and make a better fix